### PR TITLE
Add icons to Command Palette

### PR DESCRIPTION
**Pull request by janfaracik**
image

image

image

Medium-ish PR to add icons to the newly introduced Command Palette. This is one of the first refinements to search, with the intention of making search more useful and visual for users. Actions can override `getSearchIcon()` to add an icon, or if they implement `IconSpec` `getSearchIcon()` will use that by default.

Big thanks to @timja for help on this :) 

### Testing done

* Icons show in the Command Palette
* Bitmaps also show (e.g. user avatars)

### Proposed changelog entries

- Add icons to Command Palette

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@jenkinsci/sig-ux 

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/hudson/model/Computer.java
Comment (timja) [inline core/src/main/java/hudson/model/Computer.java:1115]: seems unneeded? it implements the interface already

Comment (janfaracik) [reply core/src/main/java/hudson/model/Computer.java:1115]: It implements `IComputer` which doesn't extend `IconSpec`, could extend `IconSpec` however to fix it.

Without `getSearchIcon()`:
Screenshot 2024-12-11 at 20 49 30

With `getSearchIcon()`:
image

Comment (timja) [reply core/src/main/java/hudson/model/Computer.java:1115]: Yeah just extend it in `IComputer`

**Comment by timja** [comment]
@janfaracik add the `rfe` changelog for a minor enhancement to changelog.

web-ui is just for filtering for people interested it doesn't do anything itself

**Comment by timja** [comment]
> I guess [this comment](https://github.com/jenkinsci/cloudbees-folder-plugin/pull/450/files#r1881001202) is relevant here. Why is `IconSpec` javadoc not being updated if it is no longer just for Actions/ManagementLinks?

done

**Review by timja — APPROVED** [review]
LGTM, (tested locally)

**Review by krisstern — APPROVED** [review]
Looks okay to me too!

**Comment by timja** [comment]
Looks like a test is failing consistently on this @janfaracik ?

**Comment by krisstern** [comment]
Yeah, I see the following in the logs:

```console
19:31:19  [INFO] Results:
19:31:19  [INFO] 
19:31:19  [ERROR] Failures: 
19:31:19  [ERROR]   AbstractProjectTest.autoCompleteUpstreamProjects:630->testAutoCompleteResponse:669 
19:31:19  Expected: is <true>
19:31:19       but: was <false>
19:31:19  [INFO] 
19:31:19  [ERROR] Tests run: 2212, Failures: 1, Errors: 0, Skipped: 55
19:31:19  [INFO] 
19:31:19  [ERROR] There are test failures.
19:31:19  
19:31:19  See /home/jenkins/agent/workspace/Core_jenkins_PR-10049/test/target/surefire-reports for the individual test results.
```

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by krisstern** [comment]
It looks like we also have a `NullPointerException` according to the logs at https://github.com/jenkinsci/jenkins/pull/10049/checks?check_run_id=34386281462:

```console
java.lang.NullPointerException: Cannot invoke "org.htmlunit.html.DomElement.getHtmlElementDescendants()" because "doc" is null
	at org.htmlunit.html.HtmlPage.executeDeferredScriptsIfNeeded(HtmlPage.java:1466)
	at org.htmlunit.html.HtmlPage.initialize(HtmlPage.java:265)
	at org.htmlunit.WebClient.loadWebResponseInto(WebClient.java:701)
	at org.htmlunit.WebClient.loadWebResponseInto(WebClient.java:575)
	at org.htmlunit.WebClient.getPage(WebClient.java:493)
	at org.htmlunit.WebClient.getPage(WebClient.java:402)
	at org.htmlunit.WebClient.getPage(WebClient.java:538)
	at org.htmlunit.WebClient.getPage(WebClient.java:520)
	at org.jvnet.hudson.test.JenkinsRule$WebClient.getPage(JenkinsRule.java:2735)
	at org.jvnet.hudson.test.JenkinsRule.after(JenkinsRule.java:518)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:676)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

Also one seen at https://ci.jenkins.io/job/Core/job/jenkins/job/PR-10049/7/:
```console
02:14:48  [ERROR]   FileParameterValueTest.fileParameter_canStillUse_internalHierarchy » NullPointer Cannot invoke "org.htmlunit.html.DomElement.getHtmlElementDescendants()" because "doc" is null
02:14:48  [INFO] 
02:14:48  [ERROR] Tests run: 2212, Failures: 0, Errors: 1, Skipped: 55
02:14:48  [INFO] 
02:14:48  [ERROR] 
```

File: core/src/main/java/hudson/search/Search.java
Comment (timja) [inline core/src/main/java/hudson/search/Search.java:164]: One pre-existing issue is that items with a display name are found twice, (even when the actual name is nothing alike)

image

But the display name version isn't an instanceof Item, which means it just gets the default icon.

Ideally it wouldn't report twice or at least it would provide the right item.

See pre-existing screenshot from ci.jenkins.io:
image

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
